source/main.ts

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env node
+
+// source/main.ts
+// The CLI for the `serve-handler` module.
+
+import path from 'node:path';
+import chalk from 'chalk';
+import boxen from 'boxen';
+import clipboard from 'clipboardy';
+import checkForUpdate from 'update-check';
+import manifest from '../package.json';
+import { resolve } from './utilities/promise.js';
+import { startServer } from './utilities/server.js';
+import { registerCloseListener } from './utilities/http.js';
+import { parseArguments, getHelpText } from './utilities/cli.js';
+import { loadConfiguration } from './utilities/config.js';
+import { logger } from './utilities/logger.js';
+import type { Arguments } from './types.js';
+
+/**
+ * Checks for updates to this package. If an update is available, it brings it
+ * to the user's notice by printing a message to the console.
+ *
+ * @param debugMode - Whether or not we should print additional debug information.
+ * @returns
+ */
+const printUpdateNotification = async (debugMode: boolean) => {
+  const [error, update] = await resolve(checkForUpdate(manifest));
+
+  if (error) {
+    const suffix = debugMode ? ':' : ' (use `--debug` to see full error)';
+    logger.warn(`Checking for updates failed${suffix}`);
+
+    if (debugMode) logger.error(error.message);
+  }
+  if (!update) return;
+
+  logger.log(
+    chalk` {bgRed.white  UPDATE } The latest version of \`serve\` is ${update.latest}`,
+  );
+};
+
+// Parse the options passed by the user.
+let args: Arguments;
+try {
+  args = parseArguments();
+} catch (error: any) {
+  logger.error((error as Error).message);
+  process.exit(1);
+}
+
+// Check for updates to the package unless the user sets the `NO_UPDATE_CHECK`
+// variable.
+if (process.env.NO_UPDATE_CHECK !== '1')
+  await printUpdateNotification(args['--debug']);
+// If the `version` or `help` arguments are passed, print the version or the
+// help text and exit.
+if (args['--version']) {
+  logger.log(manifest.version);
+  process.exit(0);
+}
+if (args['--help']) {
+  logger.log(getHelpText());
+  process.exit(0);
+}
+
+// Default to listening on port 3000.
+// Disabling as this is not an unnecessary check.
+// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+if (!args['--listen'])
+  args['--listen'] = [
+    [process.env.PORT ? parseInt(process.env.PORT, 10) : 3000],
+  ];
+// Ensure that the user has passed only one directory to serve.
+if (args._.length > 1) {
```

Comment:
We should make a future issue to start adding tests with `vitest` 😄

Reply:
Sure, we could add tests in a separate PR.

Reply:
https://github.com/vercel/serve/issues/709